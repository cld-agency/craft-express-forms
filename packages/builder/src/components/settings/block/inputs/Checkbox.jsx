import React from 'react';
import PropTypes from 'prop-types';
import ToggleSwitch from '../../../common/lightswitch/ToggleSwitch';
import BaseInput from './BaseInput';

class Checkbox extends BaseInput {
  static propTypes = {
    ...BaseInput.propTypes,
    value: PropTypes.bool,
  };

  toggleHandler = () => {
    const { name, value, saveHandler } = this.props;

    saveHandler({
      target: {
        name,
        value: !value,
      },
    });
  };

  renderInput() {
    const { value } = this.props;

    return <ToggleSwitch value={!!value} toggleHandler={this.toggleHandler} />;
  }
}

export default Checkbox;
