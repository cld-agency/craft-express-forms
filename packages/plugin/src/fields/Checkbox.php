<?php

namespace Solspace\ExpressForms\fields;

use craft\base\ElementInterface;
use yii\db\Schema;

class Checkbox extends BaseField
{
    public function getType(): string
    {
        return self::TYPE_CHECKBOX;
    }

    /**
     * @param $value
     *
     * @return mixed
     */
    public function normalizeValue($value, ElementInterface $element = null)
    {
        return (bool) $value;
    }

    /**
     * @param $value
     *
     * @return null|array|mixed|string
     */
    public function serializeValue($value, ElementInterface $element = null)
    {
        return (bool) $value ? '1' : '0';
    }

    /**
     * {@inheritdoc}
     */
    public function getContentColumnType(): string
    {
        return Schema::TYPE_BOOLEAN;
    }
}
