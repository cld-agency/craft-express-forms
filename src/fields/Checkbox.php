<?php

namespace Solspace\ExpressForms\fields;

use craft\base\ElementInterface;
use yii\db\Schema;

class Checkbox extends BaseField
{
    /**
     * @return string
     */
    public function getType(): string
    {
        return self::TYPE_CHECKBOX;
    }

    /**
     * @param                       $value
     * @param ElementInterface|null $element
     *
     * @return mixed
     */
    public function normalizeValue($value, ElementInterface $element = null)
    {
        return (bool) $value;
    }

    /**
     * @param                       $value
     * @param ElementInterface|null $element
     *
     * @return array|mixed|string|null
     */
    public function serializeValue($value, ElementInterface $element = null)
    {
        return (bool) $value ? '1' : '0';
    }

    /**
     * @inheritdoc
     */
    public function getContentColumnType(): string
    {
        return Schema::TYPE_BOOLEAN;
    }
}
