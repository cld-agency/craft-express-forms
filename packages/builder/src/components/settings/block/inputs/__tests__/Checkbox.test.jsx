import { shallow } from 'enzyme';
import React from 'react';
import Checkbox from '../Checkbox';

describe('<Checkbox />', () => {
  let wrapper;
  let onChange;
  beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(
      <Checkbox label="A Test checkbox field" name="test_checkbox" value={true} saveHandler={onChange} />
    );
  });

  it('renders & displays correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
