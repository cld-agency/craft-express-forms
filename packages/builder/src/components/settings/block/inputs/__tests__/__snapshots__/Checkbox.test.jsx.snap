// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders & displays correctly 1`] = `
<div
  className="block-row"
>
  <div
    className="label"
  >
    <WithStyles(ForwardRef(Tooltip))
      placement="bottom"
      title=""
    >
      <label
        className=""
      >
        A Test checkbox field
      </label>
    </WithStyles(ForwardRef(Tooltip))>
  </div>
  <div
    className="input ltr"
  >
    <ToggleSwitch
      toggleHandler={[Function]}
      value={true}
    />
  </div>
</div>
`;
